src/frameworks/compose/discover/frameworkMatcher.ts

```diff
@@ -0,0 +1,95 @@
+import { FrameworkSpec, FileSystem } from "./types";
+
+/**
+ *
+ */
+export function filterFrameworksWithDependencies(
+  allFrameworkSpecs: FrameworkSpec[],
+  dependencies: Record<string, string>
+): FrameworkSpec[] {
+  return allFrameworkSpecs.filter((framework) => {
+    return framework.requiredDependencies.every((dep) => {
+      return dep["name"] in dependencies;
+    });
+  });
+}
+
+/**
+ *
+ */
+export async function filterFrameworksWithFiles(
+  allFrameworkSpecs: FrameworkSpec[],
+  fs: FileSystem
+): Promise<FrameworkSpec[]> {
+  const filteredFrameworks = [];
+  for (const framework of allFrameworkSpecs) {
+    if (!framework.requiredFiles) {
+      filteredFrameworks.push(framework);
+      continue;
+    }
+    const isRequired = await Promise.all(
+      framework.requiredFiles.map(async (files) => {
```

Comment:
opinion: I'd just use `for` loop - I don' think we'll have more than a few files in `requireFiles` property, and I'd find a simple for loop more readable.

Understand that doing this in over a async Promise list is more performant, but it will make it much harder to debug any issue or for us to catch and tell users what's going on. (especially with nested Promise chain... that's going to have an enormous and confusing error message if anything goes wrong!)